File: 0682-baseball-game/0682-baseball-game.cpp
class Solution {
public:
    int calPoints(vector<string>& operations) {
        stack<int> Final; 

        for (int i = 0; i < operations.size(); i++){
            if (operations[i] == "+"){
                int n = Final.top(); 
                Final.pop(); 
                int m = Final.top();
                Final.push(n); 
                Final.push(m + n);  
            }else if (operations[i] == "D"){
                Final.push(2 * Final.top()); 
            }else if (operations[i] == "C"){
                Final.pop(); 
            }else {
                int num = stoi(operations[i]); 
                Final.push(num); 
            }
        }
        int Sum = 0; 
        while (!Final.empty()){
            Sum += Final.top(); 
            Final.pop();
        }
        return Sum;  
        }
        
    };

File: 0682-baseball-game/0682-baseball-game.py
class Solution:
    def calPoints(self, operations: List[str]) -> int:
        stk = []

        # +, D, C
        for i in range(len(operations)): 
            if operations[i] == "+": 
                second = int(stk[-1])
                first = 0
                if stk: 
                    first = int(stk[-2])

                Sum = first + second
                stk.append(Sum)

            elif operations[i] == "D": 
                result = stk[-1]
                stk.append(int(result)*2)
            elif operations[i] == "C": 
                stk.pop()
            else: 
                stk.append(int(operations[i]))
        
        return sum(stk)